### PR TITLE
feat(iotdb): introduced a thrift driver for IotDB connector

### PR DESCRIPTION
**Pull request by lafirest**
Fixes EMQX-12532

Release version: v/e5.8.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_connector.erl [hunk 1]
```diff
@@ -114,13 +117,13 @@ connector_example_values() ->
 namespace() -> "iotdb".
 
 roots() ->
-    [{config, #{type => hoconsc:ref(?MODULE, config)}}].
+    [].
 
-fields(config) ->
+fields("config_rest") ->
```
Comment (zmstone) [inline apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_connector.erl:122]:
```suggestion
fields("config_restapi") ->
```

File: apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_connector.erl [hunk 2]
```diff
@@ -114,13 +117,13 @@ connector_example_values() ->
...
     proplists_without(
         [url, request, retry_interval, headers],
         emqx_bridge_http_schema:fields("config_connector")
-    ) ++
+    ) ++ common_fields(rest) ++
```
Comment (zmstone) [inline apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_connector.erl:126]:
```suggestion
    ) ++ common_fields(restapi) ++
```

File: apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_connector.erl [hunk 3]
```diff
@@ -208,8 +263,13 @@ proplists_without(Keys, List) ->
...
 callback_mode() -> async_if_possible.
 
+callback_mode(#{driver := rest}) ->
+    async_if_possible;
+callback_mode(#{driver := thrift}) ->
+    always_sync.
+
 -spec on_start(manager_id(), config()) -> {ok, state()} | no_return().
-on_start(InstanceId, #{iotdb_version := Version} = Config) ->
+on_start(InstanceId, #{driver := rest, iotdb_version := Version} = Config) ->
```
Comment (zmstone) [inline apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_connector.erl:272]:
```suggestion
on_start(InstanceId, #{driver := restapi, iotdb_version := Version} = Config) ->
```

File: apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_action_info.erl [hunk 4]
```diff
@@ -53,7 +53,7 @@ bridge_v1_config_to_action_config(BridgeV1Config, ConnectorName) ->
     ).
 
 bridge_v1_config_to_connector_config(BridgeV1Config) ->
-    ConnectorKeys = schema_keys(emqx_bridge_iotdb_connector, config),
+    ConnectorKeys = schema_keys(emqx_bridge_iotdb_connector, "config_rest"),
```
Comment (zmstone) [inline apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_action_info.erl:56]:
```suggestion
    ConnectorKeys = schema_keys(emqx_bridge_iotdb_connector, "config_restapi"),
```

**Review by zmstone — COMMENTED** [review]
pleas change rest->restapi everywhere.

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-3613)

**Pull request merged**